daemon/logger/copier.go

```diff
@@ -112,7 +136,20 @@ func (c *Copier) copySrc(name string, src io.Reader) {
 					msg.Source = name
 					msg.Timestamp = time.Now().UTC()
 					msg.Line = append(msg.Line, buf[p:n]...)
-					msg.Partial = true
+
+					// Generate unique partialID for first partial. Use it across partials.
+					// Initialize index for first partial. Increment it across partials.
+					if firstPartial {
+						partialid = stringid.GenerateRandomID()
+						ordinal = 0
+						firstPartial = false
+					}
+					ordinal++
```

Comment:
Is it correct that we increment here? So a full log-entry will have `MetaData.Ordinal: 0`, and the first part of a split message will have `MetaData.Ordinal: 1`?

Reply:
Yes. A full log entry will have a Ordinal 0 and an empty ID.